### DISCUSSION TITLE
What to put in native layer?

### DISCUSSION THREAD

**Discussion by amerkoleci** [FosterFramework/Foster#23] [General]
Hi,
In my game engine I have similar layer as Foster platform, it has SDL3, stb image/truetype, rhi backends supporting vulkan/d3d12, mini audio

It uses C++ for rhi backends (specially d3d12) .

I do plan to wrap Jolt Physics and Box2D.
Question is, what would make sense to put in C++ and what in C#? High level renderer? Asset pipeline?

Thanks for suggestions!

**Comment by NoelFB** [comment]
I think the exact line is going to be subjective, but I try to just separate it with whatever ends up being easier in each language. A lot of the time stuff is much simpler in C#, so I'll put it there. But certain things like image loading, font loading, rendering, etc, is just simpler in C so it goes there. But the exact line is kind of arbitrary...

Honestly for high level rendering & asset pipelines, I would do that all in C#. I think it's performant enough (for anything I would ever make, anyway) and it will be so much easier to architect.

**Reply by amerkoleci** [reply]
Thanks for sharing your ideas, the question wasn't related to Foster and thanks for taking time to share your opinion. I find it difficult to decide which is the middle line, what need to stay where (C++ or C#) but probably I should care less and keep things going :)
